### PR TITLE
Enforce lowercasing of image and tag references.

### PR DESCRIPTION
**Pull request by xee5ch**
### Description

This modification will enforce lowercasing the image and tag name of an image before pushing. Without this, using images and tags that are not exclusively lowercase will fail to push to a standards-compliant registry. In the context of GitHub repositories, matching a GitHub repo name, that might include uppercase letters, will be used to identify the registry or image name with the repository owner or repository itself. So without this, those images will never be pushable or require manual modification.

### Related Issue(s)

Closes redhat-actions/push-to-registry#54.

### Checklist

- [ ] This PR includes a documentation change
- [x] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is not user-facing
- [x] This change is a patch change
- [ ] This change is a minor change
- [ ] This change is a major (breaking) change

### Changes made

Modified code to enforce lowercasing of image and/or tag as needed.

### CODE REVIEW

**Comment by xee5ch** [comment]
FYI, I do not know what to select re "This PR includes test changes" because there are no tests in this code base.

**Comment by tetchel** [comment]
The tests are in the workflows in `.github/workflows`, maybe we could add a job that tries to push a capitalized image name

**Comment by xee5ch** [comment]
> The tests are in the workflows in `.github/workflows`, maybe we could add a job that tries to push a capitalized image name

Sounds good, I will work on that then!

**Comment by xee5ch** [comment]
Hmmm, it seems this would be something that would also be needed to [the buildah-build GitHub Action](https://github.com/marketplace/actions/buildah-build) now that I test it more thoroughly and bumped up around errors there. :laughing: 

Should I continue with this or consider how to how to [adapt the metadata sanitization approach recommended by the Docker team with that action instead](https://github.community/t/github-actions-repository-name-must-be-lowercase/184924/6)?

**Comment by divyansh42** [comment]
@xee5ch I think we can go ahead with these changes, as this change might help a set of users who are not aware of docker metadata action and may face workflow failures.
Along with this change in Push to registry action, it will be good if we can make these changes in [buildah build](https://github.com/redhat-actions/buildah-build) action as well.

Let me know if you want to work on buildah build action too, I can create a detailed issue for that.

**Comment by xee5ch** [comment]
> @xee5ch I think we can go ahead with these changes, as this change might help a set of users who are not aware of docker metadata action and may face workflow failures. Along with this change in Push to registry action, it will be good if we can make these changes in [buildah build](https://github.com/redhat-actions/buildah-build) action as well.

Great. Thanks for supporting this!
 
> Let me know if you want to work on buildah build action too, I can create a detailed issue for that.

I can take care of it during this coming weekend, don't worry about it. :-)

> Also, please rebase your PR.

Will do, sir, yessir!

**Comment by divyansh42** [comment]
> I can take care of it during this coming weekend, don't worry about it. :-)

Great, Thanks a lot! :smile:

**Comment by xee5ch** [comment]
Sorry, I dropped the ball two weeks ago. I created redhat-actions/buildah-build#89 and will begin work on it.

**Comment by divyansh42** [comment]
@xee5ch is there any update on this PR? Thanks!

**Comment by xee5ch** [comment]
I am really sorry I dropped the ball on this. I will touch up some work and contribute it soon. Thanks for reminding me!

**Comment by divyansh42** [comment]
> I am really sorry I dropped the ball on this. I will touch up some work and contribute it soon. Thanks for reminding me!

Thanks for the update!

**Comment by divyansh42** [comment]
@xee5ch: To get this fix early, I have pushed the updated commit in your PR.
I hope you won't mind :)

File: .github/workflows/check-lowercase.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      - name: Install latest podman
+        if: matrix.install_latest
+        run: |
+          bash .github/install_latest_podman.sh
+
+      # Build image using Buildah action
+      - name: Build Image
+        id: build_image
+        # uses: redhat-actions/buildah-build@v2
+        uses: divyansh42/buildah-build@make-lowercase
```
Comment (divyansh42) [inline .github/workflows/check-lowercase.yaml:40]: Will remove this post PR https://github.com/redhat-actions/buildah-build/pull/94 merge

**Comment by xee5ch** [comment]
> @xee5ch: To get this fix early, I have pushed the updated commit in your PR. I hope you won't mind :)

I am a big flake. Yes, of course it's fine. I am sorry I didn't deliver in the demanding timeframe of six months (yes that was/is sarcasm and a joke).

I am happy you got to it before me! :laughing:

**Pull request merged**